File: DemoApp/DemoAppJava/app/src/main/java/co/optable/demoappjava/MainActivity.java
package co.optable.demoappjava;

import android.os.Bundle;
import android.util.Log;
import androidx.appcompat.app.AppCompatActivity;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.navigation.ui.AppBarConfiguration;
import androidx.navigation.ui.NavigationUI;
import co.optable.android_sdk.OptableConfig;
import co.optable.android_sdk.OptableSDK;
import com.google.android.gms.ads.MobileAds;
import com.google.android.material.bottomnavigation.BottomNavigationView;
import org.prebid.mobile.PrebidMobile;
import org.prebid.mobile.api.data.InitializationStatus;

public class MainActivity extends AppCompatActivity {

    private static final String TAG = "MainActivity";

    public static OptableSDK OPTABLE;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        OptableConfig config = new OptableConfig(this, "prebidtest", "js-sdk");
        MainActivity.OPTABLE = new OptableSDK(config);

        initGoogleAds();
        initPrebidSdk();
        initUi();
    }

    private void initGoogleAds() {
        MobileAds.initialize(this, initializationStatus -> {
        });
    }

    private void initPrebidSdk() {
        PrebidMobile.setPrebidServerAccountId("0689a263-318d-448b-a3d4-b02e8a709d9d");
        PrebidMobile.initializeSdk(getApplicationContext(), "https://prebid-server-test-j.prebid.org/openrtb2/auction", status -> {
            if (status == InitializationStatus.SUCCEEDED) {
                Log.d(TAG, "SDK initialized successfully!");
            } else {
                Log.e(TAG, "SDK initialization error: " + status.getDescription());
            }
        });
    }

    private void initUi() {
        BottomNavigationView navView = findViewById(R.id.nav_view);
        NavController navController = Navigation.findNavController(this, R.id.nav_host_fragment);
        AppBarConfiguration appBarConfiguration = new AppBarConfiguration.Builder(
                R.id.navigation_identify,
                R.id.navigation_gambanner,
                R.id.navigation_prebid
        ).build();
        NavigationUI.setupActionBarWithNavController(this, navController, appBarConfiguration);
        NavigationUI.setupWithNavController(navView, navController);
    }

}

File: DemoApp/DemoAppKotlin/app/src/main/java/co/optable/androidsdkdemo/MainActivity.kt
/*
 * Copyright © 2020 Optable Technologies Inc. All rights reserved.
 * See LICENSE for details.
 */
package co.optable.androidsdkdemo

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.findNavController
import androidx.navigation.ui.AppBarConfiguration
import androidx.navigation.ui.setupActionBarWithNavController
import androidx.navigation.ui.setupWithNavController
import co.optable.android_sdk.OptableConfig
import co.optable.android_sdk.OptableSDK
import com.google.android.gms.ads.MobileAds
import com.google.android.material.bottomnavigation.BottomNavigationView
import org.prebid.mobile.PrebidMobile

class MainActivity : AppCompatActivity() {

    companion object {
        lateinit var OPTABLE: OptableSDK

        private const val TAG = "MainActivity"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val config = OptableConfig(this, "prebidtest", "js-sdk")
        OPTABLE = OptableSDK(config)

        initGoogleAds()
        initPrebidSdk()
        initUi()
    }

    private fun initGoogleAds() {
        MobileAds.initialize(this) {}
    }

    private fun initPrebidSdk() {
        PrebidMobile.setPrebidServerAccountId("0689a263-318d-448b-a3d4-b02e8a709d9d")
        PrebidMobile.initializeSdk(
            applicationContext,
            "https://prebid-server-test-j.prebid.org/openrtb2/auction"
        ) { status: org.prebid.mobile.api.data.InitializationStatus? ->
            if (status == org.prebid.mobile.api.data.InitializationStatus.SUCCEEDED) {
                Log.d(TAG, "SDK initialized successfully!")
            } else {
                Log.e(TAG, "SDK initialization error: ${status?.description}")
            }
        }
    }

    private fun initUi() {
        val navView: BottomNavigationView = findViewById(R.id.nav_view)
        val navController = findNavController(R.id.nav_host_fragment)
        val appBarConfiguration = AppBarConfiguration(
            setOf(
                R.id.navigation_identify,
                R.id.navigation_gambanner,
            )
        )
        setupActionBarWithNavController(navController, appBarConfiguration)
        navView.setupWithNavController(navController)
    }

}